### ISSUE TITLE
tfbridge singularizes the names even when the plural names exists

### ISSUE THREAD

**Issue by mikhailshilkov** [pulumi/pulumi-terraform#490]
This is the cause for [this issue in Azure](https://github.com/pulumi/pulumi-azure/issues/243).

`azure.dns.NsRecord` resource has both `records` and `record` properties, the latter one is deprecated.

Tfbridge [singularizes](https://github.com/pulumi/pulumi-terraform/blob/master/pkg/tfbridge/names.go#L43) the name without checking whether the plural name exists in the first place.

I have a fix which first checks whether the plural name exists, and if so, skips the rest of the singilarization code. This fixes the original issue downstream.

The question is - what can this break? @lukehoban

**Comment by mikhailshilkov** [comment]
An obvious workaround is to add an explicit name mapping `records->records` to the azure provider definition. I think for now I'm going to do just that, especially because we already have a mapping for `record->record`, and making sure we pluralize correctly seems even riskier than the singularization.

**Comment by lukehoban** [comment]
Yes - I think doing a manual name mapping is the right thing for now. 

We could also add a warning or error in the code generator if a singularized name conflicts with a plural name so that we can catch this earlier (and require adding the manual mapping).
